Iss-hipm-556

BeisiXiong opened the pull request:
As a user, I want to see exhibits details only if I have already visited the exhibit.

lederr commented:
One of your droid tests is running into an endless loop which is caused by the following:

In ExhibitDetailsViewModelTest.cs the test `public void Navigation_All()` stucks at
`while (sut.NextViewAvailable)
            {
                sut.NextViewCommand.Execute(null);
                navigations++;
            }` .

I figured out, that if you leave this test out, there appears 4 more errors. All the errors seems to be related to NextViewCommand.
I guess you have to fix something in ExhibitDetailsViewModel regarding the NextViewCommand. The init looks very different in comparison to the other Commands and the Method `private async Task GotoNextView ()` should not return a Task but void (to make it look like `private async void GotoNextView ()`). If you really need a Task as a return value for reuse reasons, try to outsource this method. To show what I mean, this could look like this:
`private async void GotoNextView (){
     GotoNextView (); // The Task returning Function
}`.

Please fix this in order to make the pull request approvable ;-)

babriupb commented:
I guess the solution is much simpler, the exhibit used in the test has "Unlocked" not set therefore the value is false. Calling the NextViewCommand will result in an alert and the while loop will run infinitly long
	
So just set exhibit.Unlocked = true; in ExhibitDetailsViewModelTest.CreateSystemUnderTest and i guess it should be fixed

Resudo commented on Sources/HipMobileUI/ViewModels/Pages/ExhibitDetailsViewModel.cs:
```diff
@@ -191,22 +201,34 @@ private void ToggleVisibilityOfNavigationButtons ()
-
                 // update the UI
+            }
+            if (exhibitUnblocked)
```
Judging by the context this if-else-statement should be inside the upper if (currentViewIndex..) code segment after the statement related to the AudioPlayer. Due to NextViewAvailable it should cause no problem if it stays like this but it would be logically cleaner.

BeisiXiong commented:
thank you for your reviewing
I just revised it,and updated my PR
Please check it again.
Thank you!

On Sat 10. Jun 2017 at 18:42, Resudo <notifications@github.com> wrote:

> *@Resudo* commented on this pull request.
>
> The code is working as expected and causes no errors for me. I added one
> comment regarding a small logical issue and after changing this one thing I
> would approve the pull request.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/HiP-App/HiP-Forms/pull/84#pullrequestreview-43301373>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AZn0z6kzOgl-oFZu2GUKV8MM3qohSCIgks5sCseOgaJpZM4NosIy>
> .
>
